controllers/toolchainstatus/toolchainstatus_controller.go

```diff
@@ -412,6 +424,115 @@ func (r *Reconciler) sendToolchainStatusNotification(logger logr.Logger,
 	return nil
 }
 
+type ComponentNotReadyStatus struct {
+	ComponentType string
+	ComponentName string
+	Reason        string
+	Message       string
+}
+
+func GenerateUnreadyNotificationContent(statusMeta []ComponentNotReadyStatus) (string, error) {
+	tmpl, err := template.New("status").Parse(statusNotificationTemplate)
+	if err != nil {
+		return "", err
+	}
+
+	var output bytes.Buffer
+
+	err = tmpl.Execute(&output, statusMeta)
+	if err != nil {
+		return "", err
+	}
+
+	return output.String(), nil
+}
+
+func ExtractStatusMetadata(instance *toolchainv1alpha1.ToolchainStatus) []ComponentNotReadyStatus {
+	result := []ComponentNotReadyStatus{}
+
+	cond, found := condition.FindConditionByType(instance.Status.Conditions, toolchainv1alpha1.ConditionReady)
+	if found && cond.Status != corev1.ConditionTrue {
+		result = append(result, ComponentNotReadyStatus{
+			ComponentType: "",
+			ComponentName: "ToolchainStatus",
+			Reason:        cond.Reason,
+			Message:       cond.Message,
+		})
+	}
+
+	if instance.Status.HostOperator != nil {
+		cond, found = condition.FindConditionByType(instance.Status.HostOperator.Conditions, toolchainv1alpha1.ConditionReady)
+		if found && cond.Status != corev1.ConditionTrue {
+			result = append(result, ComponentNotReadyStatus{
+				ComponentType: "",
+				ComponentName: "Host Operator",
+				Reason:        cond.Reason,
+				Message:       cond.Message,
+			})
+		}
+	}
+
+	if instance.Status.Members != nil {
+		for _, member := range instance.Status.Members {
+			cond, found := condition.FindConditionByType(member.MemberStatus.Conditions, toolchainv1alpha1.ConditionReady)
+			if found && cond.Status != corev1.ConditionTrue {
+				result = append(result, ComponentNotReadyStatus{
+					ComponentType: "Member",
+					ComponentName: member.ClusterName,
+					Reason:        cond.Reason,
+					Message:       cond.Message,
+				})
+			}
+
+			if member.MemberStatus.Routes != nil {
+				cond, found = condition.FindConditionByType(member.MemberStatus.Routes.Conditions, toolchainv1alpha1.ConditionReady)
+				if found && cond.Status != corev1.ConditionTrue {
+					result = append(result, ComponentNotReadyStatus{
+						ComponentType: "Member Route",
+						ComponentName: member.ClusterName,
```

Comment:
Let's also print both rout URLs: Che and Console

Reply:
Done